### PR TITLE
Fix spladepp parallelism

### PR DESCRIPTION
**Pull request by joein**

### CODE REVIEW

**Comment by joein** [comment]
#170

**Review by Anush008 — APPROVED** [review]

File: tests/test_sparse_embeddings.py [hunk 1]
```diff
@@ -71,3 +71,28 @@ def test_single_embedding():
...
+
+    model = SparseTextEmbedding(
+        model_name="prithivida/Splade_PP_en_v1",
+    )
+    docs = ["hello world", "flag embedding"] * 30
+    sparse_embeddings_duo = list(model.embed(docs, batch_size=10, parallel=2))
+    sparse_embeddings_all = list(model.embed(docs, batch_size=10, parallel=0))
+    sparse_embeddings = list(model.embed(docs, batch_size=10, parallel=None))
+
+    assert len(sparse_embeddings) == len(sparse_embeddings_duo) == len(sparse_embeddings_all) == len(docs)
```
Comment (NirantK) [inline tests/test_sparse_embeddings.py:87]: I'd love it we can figure out if this parallelism actually improves throughput — and if yes, by how much?

Comment (joein) [reply tests/test_sparse_embeddings.py:87]: It is a little bit faster then builtin onnx parallelism

e.g. on my machine

processing these:

```
docs = ["hello world", "flag embedding"] * 3_000
```

with `parallel=6` took 16 seconds, while processing it with `parallel=None` took 21 seconds

Comment (joein) [reply tests/test_sparse_embeddings.py:87]: I used this code:

```python
from fastembed.sparse.sparse_text_embedding import SparseTextEmbedding


docs = ["hello world", "flag embedding"] * 3_000


if __name__ == '__main__':
    import time

    model = SparseTextEmbedding(
        model_name="prithivida/Splade_PP_en_v1"
    )
    a = time.perf_counter()
    list(model.embed(docs, batch_size=64, parallel=6))
    b = time.perf_counter()
    print('parallel = 6 ', b - a)

    c = time.perf_counter()
    list(model.embed(docs, batch_size=64, parallel=0))
    d = time.perf_counter()
    print('parallel = 0', d - c)

    list(model.embed(docs, batch_size=64, parallel=None))
    print('parallel = None', time.perf_counter() - d)
```

Comment (joein) [reply tests/test_sparse_embeddings.py:87]: The output:

```
parallel = 6  16.65197491700019
parallel = 0 17.8088272089999
parallel = None 20.514152749999994
```

**Pull request merged**